src/provider/provider_tracking.h

```diff
@@ -14,14 +14,41 @@
 #include <umf/memory_pool.h>
 #include <umf/memory_provider.h>
 
+#include "critnib.h"
+#include "utils_concurrency.h"
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 extern umf_memory_provider_ops_t UMF_TRACKING_MEMORY_PROVIDER_OPS;
 
+struct umf_memory_tracker_t {
+    critnib *map;
+    os_mutex_t *splitMergeMutex;
+};
+
 typedef struct umf_memory_tracker_t *umf_memory_tracker_handle_t;
 
+static inline umf_memory_tracker_handle_t umfMemoryTrackerCreate(void) {
+    umf_memory_tracker_handle_t handle = (umf_memory_tracker_handle_t)malloc(
+        sizeof(struct umf_memory_tracker_t));
+    if (!handle) {
+        return NULL;
+    }
+
+    handle->map = critnib_new();
+    handle->splitMergeMutex = util_mutex_create();
+
+    return handle;
+}
+
+static inline void umfMemoryTrackerDestroy(umf_memory_tracker_handle_t handle) {
+    critnib_delete(handle->map);
```

Comment:
add an assert + return if handle == NULL

Reply:
Can't return since this is done in the lib destructor anyway and we cannot handle a failure there.

Reply:
Ok but if handle == NULL then we crash when we try to deref it. Please just add an early return for this case.

Reply:
Right, good point. Done.